notebooks/02-ingestion/05-partitioning-data.ipynb

```diff
@@ -0,0 +1,717 @@
+    "  \"Cancelled\", \"CancellationCode\", \"Diverted\", \"CRSElapsedTime\", \"ActualElapsedTime\", \"AirTime\", \"Flights\", \"Distance\", \"DistanceGroup\", \"CarrierDelay\",\n",
+    "  \"WeatherDelay\", \"NASDelay\", \"SecurityDelay\", \"LateAircraftDelay\", \"FirstDepTime\", \"TotalAddGTime\", \"LongestAddGTime\", \"DivAirportLandings\", \"DivReachedDest\",\n",
+    "  \"DivActualElapsedTime\",  \"DivArrDelay\",  \"DivDistance\",  \"Div1Airport\",  \"Div1AirportID\",  \"Div1AirportSeqID\",  \"Div1WheelsOn\",  \"Div1TotalGTime\",\n",
+    "  \"Div1LongestGTime\",  \"Div1WheelsOff\",  \"Div1TailNum\",  \"Div2Airport\",  \"Div2AirportID\",  \"Div2AirportSeqID\",  \"Div2WheelsOn\",  \"Div2TotalGTime\",\n",
+    "  \"Div2LongestGTime\",  \"Div2WheelsOff\",  \"Div2TailNum\",  \"Div3Airport\",  \"Div3AirportID\",  \"Div3AirportSeqID\",  \"Div3WheelsOn\",  \"Div3TotalGTime\",\n",
+    "  \"Div3LongestGTime\",  \"Div3WheelsOff\",  \"Div3TailNum\",  \"Div4Airport\",  \"Div4AirportID\",  \"Div4AirportSeqID\",  \"Div4WheelsOn\",  \"Div4TotalGTime\",\n",
+    "  \"Div4LongestGTime\",  \"Div4WheelsOff\",  \"Div4TailNum\",  \"Div5Airport\",  \"Div5AirportID\",  \"Div5AirportSeqID\",  \"Div5WheelsOn\",  \"Div5TotalGTime\",\n",
+    "  \"Div5LongestGTime\",  \"Div5WheelsOff\",  \"Div5TailNum\"\n",
+    "FROM \"ext\"\n",
+    "PARTITIONED BY YEAR'''\n",
+    "# use a request so that we can specify the query context\n",
+    "req = sql_client.sql_request(sql)\n",
+    "req.add_context(\"rowsPerSegment\", \"50000\")  \n",
+    "display.run_task(req)\n",
+    "sql_client.wait_until_ready('flights_year_50k')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f8d8b77c-bf5d-4947-aef3-fee3c24f0ed1",
+   "metadata": {},
+   "source": [
+    "The [segment view for this table](http://localhost:8888/unified-console.html#segments/datasource~flights_year_50k) shows 12 segments that are very close to our ideal segment size. \n",
+    "\n",
+    "But let's say that the application we are building is for year long analytics for individual airlines and some airline to airline comparisons. So let's move on to clustering...\n",
+    "\n",
+    "## Clustering \n",
+    "\n",
+    "As mentioned in the introduction, time intervals of a given segment granularity can be subdivided into many segments. In the example above, we achieved this but there is no logic to the partitioning, they were just cut into 12 equally sized segments that were as close to the 50k target as possible.  \n",
+    "\n",
+    "Queries on this data will be frequently filtered on `IATA_CODE_Reporting_Airline` in order to provide individual airline analytics. So instead of just splitting the segments by quantity of rows, we can reorganize the data such that we can improve pruning when filtering on a given airline or a few airlines.\n",
```

Comment:
"When queries frequently filter on particular dimensions, CLUSTERED BY provides a way to reorganize the data, improving pruning when filtering.

Run the following cell to apply a secondary partition through the CLUSTERED BY clause on the `IATA_code_repiegeowhfewlokncoewihgowenvewovnevwoihvoewihewoh"

Reply:
See prior comment... I think this is essential for understanding when and how to use CLUSTERED BY.